commands/operator-sdk/cmd/scorecard/scorecard.go

```diff
@@ -0,0 +1,237 @@
+	cgoscheme.AddToScheme(scheme)
+	// api extensions scheme (CRDs)
+	extscheme.AddToScheme(scheme)
+	// olm api (CSVs)
+	olmApi.AddToScheme(scheme)
+	dynamicDecoder = serializer.NewCodecFactory(scheme).UniversalDeserializer()
+	// if a user creates a new CRD, we need to be able to reset the rest mapper
+	// temporary kubeclient to get a cached discovery
+	kubeclient, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to get a kubeclient: %v", err)
+	}
+	cachedDiscoveryClient := cached.NewMemCacheClient(kubeclient.Discovery())
+	restMapper = restmapper.NewDeferredDiscoveryRESTMapper(cachedDiscoveryClient)
+	restMapper.Reset()
+	runtimeClient, _ = client.New(kubeconfig, client.Options{Scheme: scheme, Mapper: restMapper})
+	err = createFromYAMLFile(SCConf.GlobalMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create global resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.NamespacedMan, false)
+	if err != nil {
+		return fmt.Errorf("failed to create namespaced resources: %v", err)
+	}
+	err = createFromYAMLFile(SCConf.CrMan, true)
+	if err != nil {
+		return fmt.Errorf("failed to create cr resource: %v", err)
+	}
+	obj := unstructured.Unstructured{}
+	obj.SetAPIVersion(apiversion)
+	obj.SetKind(kind)
+	if SCConf.BasicTests {
+		fmt.Println("Checking for existence of spec and status blocks in CR")
+		err = checkSpecAndStat(runtimeClient, obj, false)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Checking that operator actions are reflected in status")
+		err = checkStatusUpdate(runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Checking that writing into CRs has an effect")
+		logs, err := writingIntoCRsHasEffect(obj)
+		if err != nil {
+			return err
+		}
+		log.Debugf("Scorecard Proxy Logs: %v\n", logs)
+	} else {
+		err = checkSpecAndStat(runtimeClient, obj, true)
+		if err != nil {
+			return err
+		}
+	}
+	if SCConf.OlmTests {
+		yamlSpec, err := ioutil.ReadFile(SCConf.CSVPath)
+		if err != nil {
+			return fmt.Errorf("failed to read csv: %v", err)
+		}
+		rawCSV, _, err := dynamicDecoder.Decode(yamlSpec, nil, nil)
+		if err != nil {
+			return err
+		}
+		csv := &olmApi.ClusterServiceVersion{}
+		switch o := rawCSV.(type) {
+		case *olmApi.ClusterServiceVersion:
+			csv = o
+		default:
+			return fmt.Errorf("provided yaml file not of ClusterServiceVersion type")
+		}
+		fmt.Println("Checking for CRD resources")
+		crdsHaveResources(csv)
+		fmt.Println("Checking for existence CR example")
+		annotationsContainExamples(csv)
+		fmt.Println("Checking spec descriptors")
+		err = specDescriptors(csv, runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Checking status descriptors")
+		err = statusDescriptors(csv, runtimeClient, obj)
+		if err != nil {
+			return err
+		}
+	}
+	var totalEarned, totalMax int
+	for _, testType := range []string{basicOperator, olmIntegration, goodTenant} {
+		fmt.Printf("%s:\n", testType)
```

Comment:
Here's output with the minimal flags set:
```sh
$ operator-sdk scorecard --cr-manifest ./deploy/crds/example_v1alpha1_nginx_cr.yaml
Basic Operator:
OLM Integration:
Good Tentant:

Total Score: 0/0 points
```

Maybe there's a better way to output the results when test categories aren't enabled?